go/genkit/flow.go

```diff
@@ -117,26 +117,32 @@ type flowOptions struct {
 
 type noStream = func(context.Context, struct{}) error
 
-// FlowAuth configures a auth context provider and a auth policy check for a flow.
+// FlowAuth configures an auth context provider and a auth policy check for a flow.
 type FlowAuth interface {
-	// ProvideAuthContext provides auth context from an auth header.
-	ProvideAuthContext(ctx context.Context, authHeader string) (map[string]any, error)
+	// ProvideAuthContext provides auth context from an auth header and sets it on the context.
```

Comment:
There are standard-ish names for these functions; see https://pkg.go.dev/context#Context.Value.

So I would call this NewContext, since it makes a new context containing auth information.

Reply:
Full explanation in the next comment but summarize here: it doesn't just return a new context with auth information, it actually does API calls so I'm worried that using standard names like `NewContext` may be misleading in what's happening.